Chat/ClientApp/src/core/sideEffects.ts

```diff
@@ -633,6 +646,19 @@ const getToken = async () => {
   }
 };
 
+const refreshTokenAsync = async (userMri: string) => {
+  try {
+    let refreshTokenRequestOptions = {
+      method: 'GET'
+    };
+
+    let getTokenResponse = await fetch('/refreshToken/'+ userMri, refreshTokenRequestOptions);
+    return getTokenResponse.text(); // this just gets us  back our token from our service
+  } catch (error) {
+    console.error('Failed at getting token, Error: ', error);
```

Comment:
This won't throw if the fetch completed but returned an error code. Is that intentional?

Reply:
Moving to using the promise model for dealing with errors and throwing our own error if something bad happens